app/code/local/Magmodules/Webwinkelconnect/controllers/IndexController.php

```diff
@@ -48,4 +48,37 @@ public function indexAction()
         }
     }
 
+    /**
+     * @throws Exception
+     */
+    public function syncAction() {
+        $request_data = trim(file_get_contents('php://input'));
+        $helper = Mage::helper('webwinkelconnect');
+        if (!$request_data) {
+            $helper->returnResponseCode(400, 'Empty request data');
+        }
+        if (!$request_data = json_decode($request_data, true)) {
+            $helper->returnResponseCode(400, 'Invalid JSON data provided');
+        }
+
+
+        if (
+            !$helper->hasCredentialFields($request_data['webshop_id'], $request_data['api_key'])
+            || $helper->credentialsEmpty($request_data['webshop_id'], $request_data['api_key'])
+        ) {
+            $helper->returnResponseCode(403,'Missing credential fields');
+        }
+
+        $helper->isAuthorized($request_data['webshop_id'], $request_data['api_key']);
```

Comment:
* There are three functions where it could probably be just one to check the credentials.
* The response handling should stay in the controller, so just return a boolean from the helper.
* The values should be cast to string since you don't control the type in the JSON.